test/python3/libdnf/rpm/test_solv_query.py

```diff
@@ -51,6 +51,61 @@ def test_size(self):
         query = libdnf.rpm.SolvQuery(self.sack)
         self.assertEqual(query.size(), 291)
 
+    def test_iterate_packageset(self):
+        query = libdnf.rpm.SolvQuery(self.sack)
+
+        # Iterates over helping reference "pset".
+        pset = query.get_package_set()
+        prev_id = 0
+        for pkg in pset:
+            id = pkg.get_id().id
+            self.assertGreater(id, prev_id)
+            prev_id = id
+        self.assertGreaterEqual(prev_id, query.size())
+
+        # Similar to above, but longer notation.
+        pset_iterator = iter(pset)
+        prev_id = 0
+        while True:
+            try:
+                pkg = next(pset_iterator)
+                id = pkg.get_id().id
+                self.assertGreater(id, prev_id)
+                prev_id = id
+            except StopIteration:
+                break
+        self.assertGreaterEqual(prev_id, query.size())
+
+    def test_iterate_packageset2(self):
+        # Tests the iteration of an unreferenced PackageSet object. The iterator must hold a reference
+        # to the iterated object, otherwise the gargabe collector can remove the object.
+
+        query = libdnf.rpm.SolvQuery(self.sack)
+
+        # Iterates directly over "get_package_set()" result. No helping reference.
+        prev_id = 0
+        for pkg in query.get_package_set():
+            id = pkg.get_id().id
+            self.assertGreater(id, prev_id)
+            prev_id = id
+        self.assertGreaterEqual(prev_id, query.size())
```

Comment:
I believe that it works, but the comparison is incorrect. Instead of `query.size()` it should use sack size, that is represented by 'int SolvSack::Impl::get_nsolvables() { return pool->nsolvables; }; '. May be we can make get_nsolvables() as a public method. Or we can name it as size() method, but not sure. In description we have to specify what size represents.

Reply:
Adding a public method to get the number of solvables in a pool is a good idea. I added `int SolvSack::get_nsolvables() const noexcept;`. I was thinking about `size()` method but I chose the more specific name.

And I changed the tests. They are stricter now. They use `assertEqual` instead of `assertGreater` or `assertGreaterEqual`.

Reply:
LGTM